src/App.js

```diff
@@ -134,54 +138,163 @@ class App extends Component {
+      const anyNewDepressed = newlyDepressed.length > 0
+      const anyNewlyDead = newlyDead > 0
+      if (anyNewDepressed) {
+        const newlyDepressedName = newlyDepressed.name
+        newGameMessage = newlyDepressedName === 'You' ? `${newlyDepressedName} are depressed.` : `${newlyDepressedName} is depressed.`
+      }
+      if (anyNewlyDead) {
+        const newlyDeadName = newlyDead.name
+        newGameMessage = newlyDeadName === 'You'
+          ? `${newlyDeadName} have died of melancholy.`
+          : `${newlyDeadName} has died of melancholy.`
+      }
     }
-    if (newFood <= 0) {
-      gameMessage = 'You have run out of food.'
-      newFood = 'no food'
+    const youIsDead = this.isYouDead(newCharacterList) // bool
+    const newGameObj = { // gameObj changed for 'You' dead, ends game
+      gameState: youIsDead ? GAMEOVER : PLAYING, // continues or ends game depending on if 'You' in new people list is dead
+      difficulty: this.state.game.difficulty, // preserves difficulty
+      gameMessage: newGameMessage // new game message
+    }
+
+    return {
+      game: newGameObj, // hardcoded from inside function
+      progress: progressObj, // passes progressObj from walk unchanged
+      inventory: newFood, // replaces entire inventory with new food??
+      people: newCharacterList // changed people list
     }
-    this.setState({
-      progress: {miles: newMiles, days: newDays},
-      inventory: {food: newFood},
-      people: newPeopleList
-    })
+  }
+
+  starvedCharacterMessage (characterList) {
+    const deadThisPlay = characterList.filter((person) => person.status === 'dead')
+    if (deadThisPlay.length > 0) {
+      const deadPersonName = deadThisPlay[0].name
+      const message = deadPersonName === 'You'
+        ? `${deadPersonName} have died of starvation and exhaustion.`
+        : `${deadPersonName} has died of starvation and exhaustion.`
+      const newGameMessage = message
+      return newGameMessage
+    } else {
+      return ''
+    }
+  }
+
+  ignoreStatusDead () {
+    const peopleList = this.state.people.map((person) => Object.assign({}, person))
+    const peopleLiving = peopleList.filter((character) => character.status !== DEAD)
+    return peopleLiving
   }
 
   peopleLoseHealth (peopleLiving) {
-    const lostHealth = this.state.inventory.food > 0 ? 5 : 20
+    const lostHealth = this.state.inventory.food > 0 ? 5 : 20 // correct 0 to 5
     const peopleList = peopleLiving.map(function (character) { character.health -= lostHealth; return character })
-    const anyoneDead = peopleList.filter((person) => person.health <= 0)
+    return peopleList
+  }
+
+  decrementFood (newCharacterList) {
+    const inventoryObject = Object.assign({}, this.state.inventory)
+    const foodPortions = newCharacterList.length // number of people living and eating
+    const foodLost = randomGenerator(2 * foodPortions, 5 * foodPortions) // pounds eaten
```

Comment:
again, you could rename this variable `poundsEaten` instead of `foodLost` then you wouldn't need the comment.